Framework/CurveFitting/src/Functions/CubicSpline.cpp

```diff
@@ -88,22 +80,17 @@ void CubicSpline::setupInput(boost::scoped_array<double> &x,
 
     // if x[i] is out of order with its neighbours
     if (i > 1 && i < n && (x[i - 1] < x[i - 2] || x[i - 1] > x[i])) {
-      xSortFlag = true;
+      g_log.warning() << "Spline x parameters are not in ascending order. "
+                         "Only x values will be sorted.\n";
+      std::sort(x.get(), x.get() + n);
+      continue;
```

Comment:
This previous code executed the `y[i] = ` line below but this now continues. Was this intended? It would seem to me that the `y` value would get missed if a `sort` occurred.